### PR TITLE
Get recipes

### PR DESCRIPTION
**Pull request by mfaisalnazari**
Created functionality for Get recipes, created and tested models, facades,services,poros and requests.

### CODE REVIEW

File: app/controllers/api/v1/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Api::V1::RecipesController < ApplicationController
+  def index
+    n = Nutrient.find_by(id: params[:mood])
+    details = {:nutrient=> n, :cook_time=>params[:time_available]}
+    recipes = RecipeFacade.new(details).get_recipe_data
```
Comment (IgorrMagalhaess) [inline app/controllers/api/v1/recipes_controller.rb:5]: I like that you used facade to organize your code!

File: app/services/recipe_service.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    response = get_url("recipes?nutrient=#{details[:nutrient].name}&cook_time=#{details[:cook_time]}")
+  end
+
+  def get_url(url)
+    response = conn.get(url)
+    data = JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def conn
+    Faraday.new(url: "https://7a97657d-b4dd-468a-960b-563f46161622.mock.pstmn.io")
```
Comment (IgorrMagalhaess) [inline app/services/recipe_service.rb:12]: Let's leave a mental note to change this to the deployed version of this service and testing again before deploying this one.

**Review by IgorrMagalhaess — APPROVED** [review]
Love to see those tests and abstraction. Very organized. One note is maybe add /coverage/* to .gitignore.

File: spec/ models/nutrient_spec.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+
+RSpec.describe Nutrient, type: :model do
+  describe "validations" do
+    it { is_expected.to validate_presence_of(:name) }
+    it { is_expected.to validate_presence_of(:health_benefits) }
```
Comment (JackCSweeney) [inline spec/ models/nutrient_spec.rb:6]: Interesting! I don't think I've seen this syntax before for these validations! Is this a new/older syntax from shoulda-matchers? Looks good though!

File: spec/facades/recipe_facade_spec.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        recipes = facade.get_recipe_data
+
+        expect(recipes).to be_an(Array)
+        expect(recipes.size).to eq(2)
+        expect(recipes.first).to be_a(Recipe)
+        expect(recipes.last).to be_a(Recipe)
+      end
+    end
+  end
+end
```
Comment (JackCSweeney) [inline spec/facades/recipe_facade_spec.rb:69]: Super clean test here! Nicely done Faisal.

**Review by JackCSweeney — APPROVED** [review]
Looks great to me so far! Seems good to merge and can probably get started with deployment and the CI/CD stuff too!

**Pull request merged**